mods/utility/tgsv4_integration/tgs.dm

```diff
@@ -0,0 +1,378 @@
+// tgstation-server DMAPI
+
+#define TGS_DMAPI_VERSION "6.0.4"
+
+// All functions and datums outside this document are subject to change with any version and should not be relied on.
+
+// CONFIGURATION
+
+/// Create this define if you want to do TGS configuration outside of this file.
+#ifndef TGS_EXTERNAL_CONFIGURATION
+
+// Comment this out once you've filled in the below.
+//#error TGS API unconfigured
+
+// Uncomment this if you wish to allow the game to interact with TGS 3.
+// This will raise the minimum required security level of your game to TGS_SECURITY_TRUSTED due to it utilizing call()()
+//#define TGS_V3_API
+
+// Required interfaces (fill in with your codebase equivalent):
+
+/// Create a global variable named `Name` and set it to `Value`.
+#define TGS_DEFINE_AND_SET_GLOBAL(Name, Value) var/global/Name = Value
```

Comment:
`#define TGS_DEFINE_AND_SET_GLOBAL(Name, Value) GLOBAL_PROTECTED_UNTYPED(##Name, ##Value)`

Reply:
Only difference is what a global variable _should_ be protected to avoid admin interference.